docs/3.0/select-all.md

```diff
@@ -0,0 +1,43 @@
+# Select All
+
+The "Select All" feature is designed to enable users to select all queried records and perform actions on the entire selection. This feature is particularly useful when dealing with large datasets, allowing users to trigger actions on all queried records, not just the ones visible on the current table page.
+
+## How It Works
+
+<Image src="/assets/img/3_0/select_all.gif" width="687" height="289" alt="Select all demonstration" />
+
+When the "Select All" feature is activated, the actions system needs to handle the entire set of records that matches the user's query, even if only a subset of the records is currently displayed. To achieve this, the entire query is serialized and passed along with the action request.
+
+### Serialization of the Query
+
+The query might include various filters, sorting parameters, and other custom elements. Reconstructing this query at the time of the action request can be complex. Therefore, the system serializes the entire query object into a secure format before sending it with the action request.
+
+- **Security**: To ensure that sensitive data is protected, the serialized query is encrypted before it is transmitted.
+- **Efficiency**: This approach allows the system to accurately and efficiently reconstruct the original query when the action is executed, ensuring that all relevant records are included.
+
+:::warning
+<VersionReq version="3.12.0" />
+If an error occurs during the serialization process, the "Select All" feature is automatically disabled. This safeguard ensures that actions are only performed on the records currently displayed, maintaining the integrity of the application until the serialization issue can be resolved.
```

Comment:
```suggestion
If an error occurs during the serialization process, the "Select All" feature is automatically disabled. This safeguard ensures that the page will not crash because of a coding error.
We listed a few reasons on why it might crash below.
```